apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.CommonTesting/SpannerTestDatabase.cs

```diff
@@ -12,161 +12,55 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-using Google.Api.Gax;
-using Google.Api.Gax.ResourceNames;
 using Google.Cloud.ClientTesting;
-using Google.Cloud.Spanner.Admin.Instance.V1;
-using Google.Cloud.Spanner.Common.V1;
-using Google.Cloud.Spanner.V1.Internal.Logging;
-using Grpc.Core;
+using Google.Cloud.Spanner.Admin.Database.V1;
 using System;
 
-namespace Google.Cloud.Spanner.Data.CommonTesting
+namespace Google.Cloud.Spanner.Data.CommonTesting;
+
+/// <summary>
+/// A database created on-demand for testing. This is never dropped, partly as it's hard to know when to do so.
+/// (This may be used by multiple fixtures, each created and then disposed, within the same test run.)
+/// A tool is provided to clean up test databases.
+/// </summary>
+public class SpannerTestDatabase : SpannerTestDatabaseBase
 {
+    private static SpannerTestDatabase s_instance = null;
+
     /// <summary>
-    /// A database created on-demand for testing. This is never dropped, partly as it's hard to know when to do so.
-    /// (This may be used by multiple fixtures, each created and then disposed, within the same test run.)
-    /// A tool is provided to clean up test databases.
+    /// Fetches the database, creating it if necessary.
     /// </summary>
-    public class SpannerTestDatabase
+    /// <param name="projectId">The project ID to use, typically from a fixture.</param>
+    public static SpannerTestDatabase GetInstance(string projectId)
```

Comment:
Now that we are making changes here, please rename this method to Create, as Instance has Spanner specific meaning and it's confusing.
```suggestion
    public static SpannerTestDatabase Create(string projectId)
```